runtime/ops/web_worker.rs

```diff
@@ -2,40 +2,113 @@
 
 use crate::web_worker::WebWorkerHandle;
 use crate::web_worker::WorkerEvent;
+use deno_core::error::custom_error;
+use deno_core::error::generic_error;
+use deno_core::error::AnyError;
 use deno_core::futures::channel::mpsc;
+use deno_core::futures::stream::StreamExt;
+use deno_core::serde::Deserialize;
+use deno_core::serde_json;
 use deno_core::serde_json::{json, Value};
+use deno_core::ZeroCopyBuf;
+use std::sync::Arc;
+use tokio::sync::Mutex as AsyncMutex;
+
+#[derive(Deserialize)]
+struct HostUnhandledErrorArgs {
+  message: String,
+}
 
 pub fn init(
   rt: &mut deno_core::JsRuntime,
   sender: mpsc::Sender<WorkerEvent>,
-  handle: WebWorkerHandle,
+  receiver: Arc<AsyncMutex<mpsc::Receiver<Box<[u8]>>>>,
```

Comment:
Maybe the receiver could be part of `WebWorkerHandle`?

Reply:
Receiver is now part of "WorkerChannelsInternal" its the 'other end' of the communication line. WebWorkerHandle is only used to terminate the worker from within the worker.

Reply:
Normally 'WebWorkerHandle' should be only used by the host in my opinion.